tools/codegen/selective_build/selector.py

```diff
@@ -0,0 +1,112 @@
+from typing import Dict, Optional, Any, Set
+import yaml
+from abc import ABC, abstractmethod
+
+from tools.codegen.model import OperatorName
+from tools.codegen.selective_build.operator import SelectiveBuildOperator
+
+class SelectiveBuildBase(ABC):
```

Comment:
I hate to be horribly prescriptive about how exactly setup class structure, but I just wanted to briefly describe how the existing class structure, especially in `tools.codegen.model` works: the general concept is we define a bunch of frozen dataclasses representing each syntactic entity that shows up in the YAML file (e.g., there is a NativeFunction dataclass, and inside it a FunctionSchema class, etc. etc.) The modeling is very structural: the classes seek to exactly represent what you see in the input file, and then maybe we add methods and functions for doing processing on top of this.

The approach here is much more Java style, where you define an abstract base class and hide the internal data representation of the input source as an implementation detail, defining instead only a set of abstract methods by which you can interact with the data. The end result looks pretty different from the classes in, e.g., tools.codegen.model.

Let me try to argue why I think the structural approach is better than the abstract approach here. The data representation of `native_functions.yaml` and the selective build YAML files aren't internal implementation details: they are interchange formats which are relied upon by humans (in the case of `native_functions.yaml`) and machines (the producer of the selective build YAML file). You're not going to just change these formats willy-nilly, because they are themselves APIs that other people are working with: if you make a BC-breaking format change, you're going to have to edit all the producer/consumers of the format. So there isn't really any benefit to abstracting over the specifics of the data format (except in one particular case, which I'll get to shortly), because you're not going to change it. Might as well expose it properly and simplify the code (since code is simpler without abstraction).

Of course, eventually, at some point you need to make a call about whether or not an operator is included or not. So you do need some abstraction here. But I don't think you need OO-style abstraction here: a simple function that cases over the various format types would be good enough.

WDYT?